platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/EventDurabilityNexus.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.preconsensus;
+
+import com.swirlds.common.threading.CountUpLatch;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * A class used to determine if an event is guaranteed to be durable, i.e. flushed to disk.
+ */
+public class EventDurabilityNexus {
+    /**
+     * The highest event sequence number that has been flushed durably to disk.
+     */
+    private final CountUpLatch latestDurableSequenceNumber = new CountUpLatch(-1);
+
+    /**
+     * Set the highest event sequence number that has been flushed.
+     *
+     * @param lastFlushedEvent the highest event sequence number that has been flushed
+     */
+    public void setLatestDurableSequenceNumber(final long lastFlushedEvent) {
+        this.latestDurableSequenceNumber.set(lastFlushedEvent);
+    }
+
+    /**
+     * Determine if an event is guaranteed to be durable, i.e. flushed to disk
+     *
+     * @param event the event in question
+     * @return true if the event is guaranteed to be durable, false otherwise
+     */
+    public boolean isEventDurable(@NonNull final GossipEvent event) {
+        Objects.requireNonNull(event);
+        if (event.getStreamSequenceNumber() == GossipEvent.STALE_EVENT_STREAM_SEQUENCE_NUMBER) {
+            // Stale events are not written to disk.
+            return false;
+        }
```

Comment:
Not necessarily something we need to do in this PR: setting a different sequence number for stale events is overly complex and not really necessary. Eventually I'd like to stop doing it.